addons/controllers/antreaconfig_controller_test.go

```diff
@@ -207,6 +211,145 @@ var _ = Describe("AntreaConfig Reconciler and Webhooks", func() {
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.AntreaIPAM).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.ServiceExternalIP).Should(Equal(false))
+				Expect(config.Spec.Antrea.AntreaConfigDataValue.FeatureGates.Multicast).Should(Equal(false))
+				Expect(config.Spec.AntreaNsx.Enable).Should(Equal(true))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				cluster := &clusterapiv1beta1.Cluster{}
+				err := k8sClient.Get(ctx, clusterKey, cluster)
+				if err != nil {
+					return false
+				}
+
+				serviceCIDR, serviceCIDRv6, err := util.GetServiceCIDRs(cluster)
+				if err != nil {
+					return false
+				}
+
+				infraProvider, err := util.GetInfraProvider(cluster)
+				if err != nil {
+					return false
+				}
+
+				// Check infraProvider values
+				Expect(infraProvider).Should(Equal("vsphere"))
+
+				// Check ServiceCIDR and ServiceCIDRv6 values
+				Expect(serviceCIDR).Should(Equal("192.168.0.0/16"))
+				Expect(serviceCIDRv6).Should(Equal("fd00:100:96::/48"))
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				secretKey := client.ObjectKey{
+					Namespace: clusterNamespace,
+					Name:      util.GenerateDataValueSecretName(clusterName, constants.AntreaAddonName),
+				}
+				secret := &v1.Secret{}
+				err := k8sClient.Get(ctx, secretKey, secret)
+				if err != nil {
+					return false
+				}
+				Expect(secret.Type).Should(Equal(v1.SecretTypeOpaque))
+
+				// check data value secret contents
+				secretData := string(secret.Data["values.yaml"])
+
+				Expect(strings.Contains(secretData, "serviceCIDR: 192.168.0.0/16")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "serviceCIDRv6: fd00:100:96::/48")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "infraProvider: vsphere")).Should(BeTrue())
+
+				Expect(strings.Contains(secretData, "trafficEncapMode: encap")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "tlsCipherSuites: TLS_ECDHE_ECDSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_128_GCM_SHA256,TLS_ECDHE_RSA_WITH_AES_256_GCM_SHA384,TLS_ECDHE_ECDSA_WITH_AES_256_GCM_SHA384,TLS_RSA_WITH_AES_256_GCM_SHA384")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "AntreaProxy: true")).Should(BeTrue())
+				Expect(strings.Contains(secretData, "AntreaPolicy: true")).Should(BeTrue())
+
+				return true
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			Eventually(func() bool {
+				// Check status.secretRef after reconciliation
+				config := &cniv1alpha1.AntreaConfig{}
+				err := k8sClient.Get(ctx, configKey, config)
+				if err != nil {
+					return false
+				}
+				return config.Status.SecretRef == util.GenerateDataValueSecretName(clusterName, constants.AntreaAddonName)
+			}, waitTimeout, pollingInterval).Should(BeTrue())
+
+			// check if ProviderServiceAccount is created
+			By("create ProviderServiceAccount")
```

Comment:
what about "check if ProviderServiceAccount is created"

Reply:
fixed